spikeinterface/qualitymetrics/quality_metric_calculator.py

```diff
@@ -49,11 +45,16 @@ def _set_params(self, metric_names=None, qm_params=None, peak_sign=None,
         if metric_names is None:
             metric_names = list(_misc_metric_name_to_func.keys())
             # if PC is available, PC metrics are automatically added to the list
-            if self.principal_component is not None:
+            if self.waveform_extractor.is_extension('principal_components'):
                 # by default 'nearest_neightbor' is removed because too slow
                 pc_metrics = _possible_pc_metric_names.copy()
                 pc_metrics.remove("nearest_neighbor")
                 metric_names += pc_metrics
+            # if spike_locations are not available, drift is removed from the list
+            if not self.waveform_extractor.is_extension('spike_locations'):
+                if "drift" in metric_names:
+                    metric_names.remove("drift")
+
```

Comment:
I am not super happy with this hard coded. Stuff.
I known we did it for PCA. We should find something more elegant.
We can keep it at the moment.